From f9095b5d01c8e212ed39d5b0eabdd319db18ea47 Mon Sep 17 00:00:00 2001
From: Matt Hippely <mhippely@vmware.com>
Date: Thu, 30 Jul 2020 17:20:36 -0700
Subject: [PATCH] docs: update home page responsiveness

- simplified transition between hero and content
- removed the spacer / angle / triangle between whats new and content
  below - not sure how to extend it to the edge of screen on very wide
screens

Signed-off-by: Matt Hippely <mhippely@vmware.com>
---
 apps/website/.vuepress/config.js              |  16 +-
 .../.vuepress/theme/components/Footer.vue     |   4 +-
 .../.vuepress/theme/components/Home.vue       | 250 +++++++++---------
 .../theme/global-components/WhatsNew.vue      |   2 +-
 apps/website/.vuepress/theme/styles/home.scss |  13 -
 .../website/.vuepress/theme/styles/index.scss |   1 -
 .../.vuepress/theme/styles/overrides.scss     |   8 +
 7 files changed, 144 insertions(+), 150 deletions(-)
 delete mode 100644 apps/website/.vuepress/theme/styles/home.scss

diff --git a/apps/website/.vuepress/config.js b/apps/website/.vuepress/config.js
index 02754fc47..cd71b909b 100644
--- a/apps/website/.vuepress/config.js
+++ b/apps/website/.vuepress/config.js
@@ -91,15 +91,15 @@ module.exports = {
     ],
     ...blocks,
   ],
-  // devServer: {
-    // proxy: {
+  devServer: {
+    proxy: {
       // Local dev server needs to proxy to CDN too
-      // '/images': {
-      //   target: 'http://dt7zex2d2lk4u.cloudfront.net',
-      //   changeOrigin: true,
-      // },
-    // },
-  // },
+      '/images': {
+        target: 'http://dt7zex2d2lk4u.cloudfront.net',
+        changeOrigin: true,
+      },
+    },
+  },
   configureWebpack: {
     resolve: {
       alias: {
diff --git a/apps/website/.vuepress/theme/components/Footer.vue b/apps/website/.vuepress/theme/components/Footer.vue
index 7e7e807af..2d8e24ede 100644
--- a/apps/website/.vuepress/theme/components/Footer.vue
+++ b/apps/website/.vuepress/theme/components/Footer.vue
@@ -49,9 +49,9 @@ export default {
   &:before {
     content: '';
     position: absolute;
-    width: calc(100% + 2.4rem);
+    width: 100%;
     top: 0;
-    left: -1.2rem;
+    left: 0;
     border-top: 1px solid #ccc;
   }
 }
diff --git a/apps/website/.vuepress/theme/components/Home.vue b/apps/website/.vuepress/theme/components/Home.vue
index f7805660b..14afdac46 100644
--- a/apps/website/.vuepress/theme/components/Home.vue
+++ b/apps/website/.vuepress/theme/components/Home.vue
@@ -5,11 +5,11 @@
 </style>
 <template>
   <!--  <div class="home-wrapper">-->
-  <main class="page-home">
-    <div class="section-wrapper" cds-layout="vertical gap:md gap@sm:xxl">
-      <!-- Hero -->
-      <section class="home-hero" cds-layout="vertical align:horizontal-center">
-        <h1 cds-text="display center semibold" cds-layout="m-t:x">Meet Clarity</h1>
+  <main class="home-wrapper" cds-layout="vertical align:horizontal-center gap:md gap@sm:xl">
+    <!-- hero -->
+    <div class="section-content hero-wrapper" cds-layout="horizontal align:horizontal-center p-t:md p-t@sm:xxl">
+      <div class="hero" cds-layout="vertical align:horizontal-center p:md">
+        <h1 cds-text="display center semibold">Meet Clarity</h1>
         <div cds-layout="grid cols@sm:10 cols@xs:12">
           <div cds-layout="col@sm:start-2">
             <h2 cds-text="title center medium" cds-layout="m-t:xl m-x@sm:sm">
@@ -22,151 +22,151 @@
           <!-- hard coded for now, css overrides were not working. Anywhere. -->
           <router-link style="color: #000;" class="hero-btn-link" to="/get-started">Get Started</router-link>
         </button>
-        <img aria-hidden="true" cds-layout="m-t:xl container:fill" src="/images/home/img-hero.svg" />
-      </section>
-      <!-- image spacer -->
-      <section class="hero-spacer" cds-layout="horizontal align:horizontal-stretch m-t:none">
-        <img cds-layout="container:fill" aria-hidden="true" src="/images/home/bg-graphic1-fullwidth.svg" alt="" />
-      </section>
-      <!-- whats new -->
-      <section class="whats-new" cds-layout="grid gap:sm">
-        <div cds-layout="col:start-2 col:10">
-          <WhatsNew version="latest" />
+        <img aria-hidden="true" cds-layout="container:fill" src="/images/home/img-hero.svg" />
+      </div>
+    </div>
+    <!-- whats new -->
+    <section class="section-content whats-new" cds-layout="grid">
+      <div cds-layout="col:start-2 col:10">
+        <WhatsNew version="latest" />
+      </div>
+      <div cds-layout="col:12">
+        <div cds-layout="grid cols:auto align:horizontal-center">
+          <cds-button class="btn-link" action="outline" cds-layout="m-t:xl m-b:sm m-b:md">
+            <a href="/releases/v4">View Details</a>
+          </cds-button>
         </div>
+      </div>
+    </section>
+    <!-- design+dev section -->
+    <section class="section-content" cds-layout="vertical m-t:none">
+      <h3 cds-text="display center semibold" cds-layout="m-t:lg m-t@md:xxl">
+        Build inclusive and consistent products with Clarity
+      </h3>
+      <div cds-layout="grid">
         <div cds-layout="col:12">
-          <div cds-layout="grid cols:auto align:horizontal-center">
-            <cds-button class="btn-link" action="outline" cds-layout="m-t:xl m-b:sm m-b:md">
-              <a href="/releases/v4">View Details</a>
-            </cds-button>
-          </div>
-        </div>
-      </section>
-      <!-- spacer image -->
-      <section class="section-spacer" cds-layout="m:none">
-        <img cds-layout="container:fill" aria-hidden="true" src="/images/home/bg-graphic2-L-align.svg" />
-      </section>
-      <!-- design+dev section -->
-      <section>
-        <h3 cds-text="display center semibold" cds-layout="m-b:xl">
-          Build inclusive and consistent products with Clarity
-        </h3>
-        <div cds-layout="grid cols@sm:10 cols@xs:12">
-          <div cds-layout="col@sm:start-2">
-            <div cds-layout="m-t:xl grid gap:sm cols@sm:6 cols@xs:12 ">
-              <div class="image-cell" cds-layout="horizontal align:right">
-                <img cds-layout="container:fill" src="/images/home/img-design-dev.svg" aria-hidden="true" />
-              </div>
-              <div class="content-cell" cds-layout="vertical gap:sm align:center m@xs:xs">
-                <h4 cds-text="heading" cds-layout="p-b:lg">Design, Development and Deploy</h4>
-                <p class="section-block" cds-text="message">
-                  Our design guidelines, component library, and front end assets are built by working directly within
-                  product teams. Use Clarity to enable your developers and designers to deliver the best experience
-                  possible to your users.
-                </p>
-              </div>
+          <div cds-layout="m-t:xl grid gap:lg cols@sm:6 cols@xs:12 ">
+            <div class="image-cell" cds-layout="horizontal align:right">
+              <img cds-layout="container:fill" src="/images/home/img-design-dev.svg" aria-hidden="true" />
             </div>
-          </div>
-        </div>
-      </section>
-      <!-- a11y section -->
-      <section>
-        <div cds-layout="grid cols@sm:10 cols@xs:12">
-          <div cds-layout="col@sm:start-2">
-            <div cds-layout="grid gap:sm cols@sm:6 cols@xs:12 ">
-              <div class="content-cell" cds-layout="vertical gap:sm align:center m@xs:xs">
-                <h4 cds-text="heading" cds-layout="p-b:lg">Accessible</h4>
-                <p class="section-block" cds-text="message">
-                  We are continually investing heavily in accessibility. Use Clarity to address the needs of a large and
-                  diverse user base.
-                </p>
-              </div>
-              <div class="image-cell" cds-layout="horizontal align:right">
-                <img cds-layout="container:fill" src="/images/home/img-accessibility.svg" aria-hidden="true" />
-              </div>
+            <div class="content-cell" cds-layout="vertical gap:sm align:center m:sm m@sm:lg">
+              <h4 cds-text="heading" cds-layout="p-b:lg">Design, Develop and Deploy</h4>
+              <p class="section-block" cds-text="message">
+                Our design guidelines, component library, and front end assets are built by working directly within
+                product teams. Use Clarity to enable your developers and designers to deliver the best experience
+                possible to your users.
+              </p>
             </div>
           </div>
         </div>
-      </section>
-      <!-- enterprise section -->
-      <section>
-        <div cds-layout="grid cols@sm:10 cols@xs:12">
-          <div cds-layout="col@sm:start-2">
-            <div cds-layout="grid gap:sm cols@sm:6 cols@xs:12 ">
-              <div class="image-cell" cds-layout="horizontal align:right">
-                <img cds-layout="container:fill" src="/images/home/img-enterprise-ready.svg" aria-hidden="true" />
-              </div>
-              <div class="content-cell" cds-layout="vertical gap:sm align:center m@xs:xs">
-                <h4 cds-text="heading" cds-layout="p-b:lg">Enterprise Ready</h4>
-                <p class="section-block" cds-text="message">
-                  Many Fortune 500 companies rely on us to support demanding, high-performance environments. Use Clarity
-                  for your high stakes, data intensive applications.
-                </p>
-              </div>
-            </div>
+      </div>
+    </section>
+    <!-- a11y section -->
+    <section class="section-content" cds-layout="grid">
+      <div cds-layout="col:12">
+        <div cds-layout="m-t:xl grid gap:lg cols@sm:6 cols@xs:12 ">
+          <div class="content-cell" cds-layout="vertical gap:sm align:center m:sm m@sm:lg">
+            <h4 cds-text="heading" cds-layout="p-b:lg">Accessible</h4>
+            <p class="section-block" cds-text="message">
+              We are continually investing heavily in accessibility. Use Clarity to address the needs of a large and
+              diverse user base.
+            </p>
+          </div>
+          <div class="image-cell" cds-layout="horizontal align:right">
+            <img cds-layout="container:fill" src="/images/home/img-accessibility.svg" aria-hidden="true" />
           </div>
         </div>
-      </section>
-      <!-- open source section -->
-      <section>
-        <div cds-layout="grid cols@sm:10 cols@xs:12">
-          <div cds-layout="col@sm:start-2">
-            <div cds-layout="grid gap:sm cols@sm:6 cols@xs:12 ">
-              <div class="content-cell" cds-layout="vertical gap:sm align:center m@xs:xs">
-                <h4 cds-text="heading" cds-layout="p-b:lg">Open Source</h4>
-                <p class="section-block" cds-text="message">
-                  Our vision since inception has been focused on supporting a community of designers and developers
-                  working together. Clarity is supported by the community through 15k downloads a week.
-                </p>
-              </div>
-              <div class="image-cell" cds-layout="horizontal align:right">
-                <img cds-layout="container:fill" src="/images/home/img-open-source.svg" aria-hidden="true" />
-              </div>
-            </div>
+      </div>
+    </section>
+    <!-- enterprise section -->
+    <section class="section-content" cds-layout="grid">
+      <div cds-layout="col:12">
+        <div cds-layout="m-t:xl grid gap:lg cols@sm:6 cols@xs:12 ">
+          <div class="image-cell" cds-layout="horizontal align:right">
+            <img cds-layout="container:fill" src="/images/home/img-design-dev.svg" aria-hidden="true" />
+          </div>
+          <div class="content-cell" cds-layout="vertical gap:sm align:center m:sm m@sm:lg">
+            <h4 cds-text="heading" cds-layout="p-b:lg">Enterprise Ready</h4>
+            <p class="section-block" cds-text="message">
+              Many Fortune 500 companies rely on us to support demanding, high-performance environments. Use Clarity for
+              your high stakes, data intensive applications.
+            </p>
           </div>
         </div>
-      </section>
-      <!-- resources section -->
-      <section>
-        <h3 cds-text="display center medium" cds-layout="m-b:xl">Popular Resources</h3>
-        <div cds-layout="grid cols@sm:10 cols@xs:12">
-          <div cds-layout="col@sm:start-2">
-            <div cds-layout="m-y:xl grid gap:sm cols@sm:6 cols@xs:12">
-              <div class="image-cell" cds-layout="m-b:xxl m-b@xs:none horizontal align:left">
-                <img cds-layout="container:fill" src="/images/home/img-pr-components.svg" aria-hidden="true" />
-                <h4 cds-text="title medium" cds-layout="m-t:lg">Components</h4>
-                <p cds-text="message medium" cds-layout="m-t:md">
-                  Access UX guidelines and developer documentation for interactive components.
-                </p>
-              </div>
-              <div class="image-cell" cds-layout="horizontal align:left m-t@xs:sm m-t@sm:none">
-                <img cds-layout="container:fill" src="/images/home/img-pr-icons.svg" aria-hidden="true" />
-                <h4 cds-text="title medium" cds-layout="m-t:lg">Icons</h4>
-                <p cds-text="message medium" cds-layout="m-t:md">
-                  Over 380 flexible and easy to use SVG icons, complete with variations to meet your design needs.
-                </p>
-              </div>
-            </div>
+      </div>
+    </section>
+    <!-- open source section -->
+    <section class="section-content" cds-layout="grid">
+      <div cds-layout="col:12">
+        <div cds-layout="m-t:xl grid gap:lg cols@sm:6 cols@xs:12 ">
+          <div class="content-cell" cds-layout="vertical gap:sm align:center m:sm m@sm:lg">
+            <h4 cds-text="heading" cds-layout="p-b:lg">Open Source</h4>
+            <p class="section-block" cds-text="message">
+              Our vision since inception has been focused on supporting a community of designers and developers working
+              together. Clarity is supported by the community through 15k downloads a week.
+            </p>
+          </div>
+          <div class="image-cell" cds-layout="horizontal align:right">
+            <img cds-layout="container:fill" src="/images/home/img-open-source.svg" aria-hidden="true" />
           </div>
         </div>
-      </section>
-    </div>
-    <section cds-layout="m-l:lg m-b:lg">
-      <Footer></Footer>
+      </div>
     </section>
+    <!-- resources section -->
+    <section class="section-content" cds-layout="vertical">
+      <h3 cds-text="display center medium" cds-layout="m-b:xl">Popular Resources</h3>
+      <div cds-layout="grid gap:lg cols@xs:12 cols@sm:6">
+        <div class="image-cell" cds-layout="m-b:xxl m-b@xs:none">
+          <img cds-layout="container:fill" src="/images/home/img-pr-components.svg" aria-hidden="true" />
+          <h4 cds-text="title medium" cds-layout="m-t:lg">Components</h4>
+          <p cds-text="message medium" cds-layout="m-t:md">
+            Access UX guidelines and developer documentation for interactive components.
+          </p>
+        </div>
+        <div class="image-cell" cds-layout="m-b:xxl m-b@xs:none">
+          <img cds-layout="container:fill" src="/images/home/img-pr-icons.svg" aria-hidden="true" />
+          <h4 cds-text="title medium" cds-layout="m-t:lg">Icons</h4>
+          <p cds-text="message medium" cds-layout="m-t:md">
+            Over 380 flexible and easy to use SVG icons, complete with variations to meet your design needs.
+          </p>
+        </div>
+      </div>
+    </section>
+    <Footer class="footer"></Footer>
   </main>
   <!--  </div>-->
 </template>
 
-<style>
+<style lang="scss">
 .content-cell,
 image-cell {
   height: 100%;
 }
 
-.page-home {
+.section-content {
   max-width: 960px;
 }
 
+.footer {
+  width: 100%;
+}
+
+.hero-wrapper {
+  background-color: #002f44;
+  width: 100%;
+  clip-path: polygon(0 0, 100% 0%, 100% calc(100% - 10rem), 0% 100%);
+  color: #ffffff;
+  padding-bottom: 10rem;
+  max-width: unset;
+  @media only screen and (max-width: 576px) {
+    clip-path: polygon(0 0, 100% 0%, 100% calc(100% - 5rem), 0% 100%);
+    padding-bottom: 5rem;
+  }
+  .hero {
+    max-width: 960px;
+  }
+}
+
 .hero-btn {
   background-color: hsl(198, 69%, 69%);
 }
diff --git a/apps/website/.vuepress/theme/global-components/WhatsNew.vue b/apps/website/.vuepress/theme/global-components/WhatsNew.vue
index f718a7115..0c1564762 100644
--- a/apps/website/.vuepress/theme/global-components/WhatsNew.vue
+++ b/apps/website/.vuepress/theme/global-components/WhatsNew.vue
@@ -1,5 +1,5 @@
 <template>
-  <div cds-layout="vertical gap:md m-t: lg">
+  <div cds-layout="vertical gap:md m-t:lg">
     <h3 cds-text="heading medium" cds-layout="m-b:xl">What's New</h3>
     <p cds-text="title semibold">
       {{ data.version }}
diff --git a/apps/website/.vuepress/theme/styles/home.scss b/apps/website/.vuepress/theme/styles/home.scss
deleted file mode 100644
index e6ad9c413..000000000
--- a/apps/website/.vuepress/theme/styles/home.scss
+++ /dev/null
@@ -1,13 +0,0 @@
-// Overrides for homepage b/c these don't apply when in the component stylesheet.
-.home-hero {
-  padding-top: 2rem;
-  background-color: #002f44;
-  // we need dark theme colors for this hero section
-  * {
-    color: var(--cds-token-color-neutral-0);
-  }
-
-  h2 {
-    max-width: 50rem;
-  }
-}
diff --git a/apps/website/.vuepress/theme/styles/index.scss b/apps/website/.vuepress/theme/styles/index.scss
index 7fd203d57..10c345103 100644
--- a/apps/website/.vuepress/theme/styles/index.scss
+++ b/apps/website/.vuepress/theme/styles/index.scss
@@ -6,4 +6,3 @@
 @import './containers';
 @import './spaces';
 @import './layout';
-@import './home';
diff --git a/apps/website/.vuepress/theme/styles/overrides.scss b/apps/website/.vuepress/theme/styles/overrides.scss
index a35d7eeff..6e9a3d1b3 100644
--- a/apps/website/.vuepress/theme/styles/overrides.scss
+++ b/apps/website/.vuepress/theme/styles/overrides.scss
@@ -163,3 +163,11 @@ h4 .header-anchor cds-icon {
   width: var(--cds-token-space-size-9);
   height: var(--cds-token-space-size-9);
 }
+
+.hero {
+  h1,
+  h2,
+  p {
+    color: #ffffff;
+  }
+}
-- 
2.24.3 (Apple Git-128)

